test/core/blockchain/production/block_producer_test.cpp

```diff
@@ -0,0 +1,182 @@
+  return Address{Address::makeFromId(id)};
+}
+
+BlockProducerTest::Tipset BlockProducerTest::getParentTipset() const {
+  std::vector<CID> cids{decodeCID(config::kParentTipsetBlocks[0]),
+                        decodeCID(config::kParentTipsetBlocks[1])};
+  return Tipset{.cids = std::move(cids), .blks = {}, .height = 0};
+}
+
+BlockProducerTest::EPostProof BlockProducerTest::getEPostProof() const {
+  fc::common::Buffer proof_bytes{std::vector<uint8_t>(
+      config::kPostProof.begin(), config::kPostProof.end())};
+  PostRandomness randomness{};
+  std::copy_n(config::kPostRand.begin(), randomness.size(), randomness.begin());
+  return EPostProof{
+      .proof = {proof_bytes}, .post_rand = {randomness}, .candidates = {}};
+}
+
+BlockProducerTest::Ticket BlockProducerTest::getTicket() const {
+  fc::crypto::vrf::VRFProof ticket_proof{};
+  std::copy_n(
+      config::kTicket.begin(), ticket_proof.size(), ticket_proof.begin());
+  return Ticket{.bytes = ticket_proof};
+}
+
+std::vector<BlockProducerTest::SignedMessage>
+BlockProducerTest::getSampleMessages() const {
+  using fc::vm::actor::MethodNumber;
+  using fc::vm::actor::MethodParams;
+  UnsignedMessage message_A{getAddress(1),
+                            getAddress(2),
+                            123,
+                            5,
+                            100,
+                            1000,
+                            MethodNumber{1},
+                            MethodParams{}};
+  UnsignedMessage message_B{getAddress(2),
+                            getAddress(3),
+                            456,
+                            10,
+                            120,
+                            800,
+                            MethodNumber{2},
+                            MethodParams{}};
+  auto signature_A_bytes{
+      "6162636465666768696a6b6c6d6e6f707172737475767778797a6162636465666768696a"
+      "6b6c6d6e6f7071"
+      "72737475767778797a6162636465666768696a6b6c6d6e6f707172737475767778797a61"
+      "62636465666768"
+      "696a6b6c6d6e6f707172"_unhex};
+  auto signature_B{
+      "7271706f6e6d6c6b6a6968676665646362617a797877767574737271706f6e6d6c6b6a69"
+      "68676665646362"
+      "617a797877767574737271706f6e6d6c6b6a6968676665646362617a7978777675747372"
+      "71706f6e6d6c6b"
+      "6a696867666564636261"_unhex};
+  BlsSignature signature_A{};
+  std::copy_n(
+      signature_A_bytes.begin(), signature_A_bytes.size(), signature_A.begin());
+  return {{message_A, signature_A}, {message_B, signature_B}};
+}
+
+/**
+ * @given Sample data and required modules
+ * @when Generating new block
+ * @then Operation must be completed successfully
+ */
+TEST_F(BlockProducerTest, Main) {
+  /**
+   * Setup IPFS datastore, which should contain parent tipset
+   * BlockProducer will try to obtain serialized parent tipset bytes by CID
+   */
+  auto parent_tipset = getParentTipset();
+  EXPECT_OUTCOME_TRUE(parent_tipset_raw_bytes,
```

Comment:
Ipld store is primitive, why not use in-memory store, and put `setCbor(parent_tipset)` instead of mocking